### ISSUE TITLE
SPFX and Angular elements: Out of stack space error in Internet Explorer 11

### ISSUE THREAD

**Issue by Basselt** [SharePoint/sp-dev-docs#4077]
#### Category
- [X ] Question
- [ ] Typo
- [X ] Bug
- [ ] Additional article idea

I am trying to integrate Angular Elements with SPFX. 
I followed the steps in the following post https://developer.microsoft.com/en-us/office/blogs/announcing-angular-elements-support-in-pnp-spfx-generator/ 

After a lot of hours/days to fix the issue into IE 11 , I wasn't able to show a simple angular element in Workbench or SharePoint 2019. It shows only in Chrome. 

In Internet explorer I received the below error message in the observed behavior:

I tried to fix with a lot of combination of polyfills but nothing worked with me.

Can anyone help since it is very frustrating.

#### Expected or Desired Behavior

The expected behavior is to show a simple angular element usnig SPFX

#### Observed Behavior

Here are the error message I receive in Internet Explorer 11:
ERROR:
**Out of stack space**

CALL STACK:
Error: Out of stack space
   at Collection (http://localhost:3300/node_modules/@microsoft/sp-loader/dist/sp-loader-assembly_en-us.js:436:7)
   at Collection (http://localhost:3300/node_modules/@microsoft/sp-loader/dist/sp-loader-assembly_en-us.js:436:53)
   at Collection (http://localhost:3300/node_modules/@microsoft/sp-loader/dist/sp-loader-assembly_en-us.js:436:53)
   at Collection (http://localhost:3300/node_modules/@microsoft/sp-loader/dist/sp-loader-assembly_en-us.js:436:53)

**Comment by andrewconnell** [comment]
Sounds like a memory issue... can you provide more context as to how you're building the Angular project, the size of the generated library you're loading, etc? Verify other browsers render ok?

While a new version of Angular recently shipped (v8), I'm not sure if their new rendering engine (Ivy) made it in nor what sort of impact it has on the page payload (early on it was expected to shrink the size of the generated bundle dramatically).

**Comment by Basselt** [comment]
> Sounds like a memory issue... can you provide more context as to how you're building the Angular project, the size of the generated library you're loading, etc? Verify other browsers render ok?
> 
> While a new version of Angular recently shipped (v8), I'm not sure if their new rendering engine (Ivy) made it in nor what sort of impact it has on the page payload (early on it was expected to shrink the size of the generated bundle dramatically).

Thank you for your quick reply. Actually I have been trying Since February with Angular 6 and I never had it working in Internet Explorer. In Chrome it works perfectly. and other browsers too.
Only the problem seems with IE. Last week I scaffold the project using yo @pnp/spfx
So basically I am using the command:
ng build --prod --output-hashing none && node elements-build.js

The output of this is bundle.js loaded into SPFX. The size of bundle.js is 287 KB. I don't think this is large enough.

I appreciate your help and guidance.

Now I am using the

**Comment by andrewconnell** [comment]
I'm not surprised... IE11 is not good with any of the modern capabilities. While there is a polyfill for custom elements, IE11 has all sorts of issues hence why MSFT is moving forward to Edge Chromium. Not the answer you're looking for, but you're really pushing the boundary of what IE11 can do... 

Have you been able to get Angular Elements working in IE11 _WITHOUT_ SPFx (ie: just a basic page)? There's nothing special about SPFx here... but I suspect under the covers, IE11 is simply failing loading all the polyfills (with systemjs) SPFx needs + what Angular Elements needs.

**Comment by Basselt** [comment]
> I'm not surprised... IE11 is not good with any of the modern capabilities. While there is a polyfill for custom elements, IE11 has all sorts of issues hence why MSFT is moving forward to Edge Chromium. Not the answer you're looking for, but you're really pushing the boundary of what IE11 can do...
> 
> Have you been able to get Angular Elements working in IE11 _WITHOUT_ SPFx (ie: just a basic page)? There's nothing special about SPFx here... but I suspect under the covers, IE11 is simply failing loading all the polyfills (with systemjs) SPFx needs + what Angular Elements needs.

Hi Andrew,
Yes I was able to make Angular Element work by itself in IE. Only when I loaded into SPFX, it won't let me. But even if not the answer I would hope, but at least I won't waste more time on it until further improvement.

Thanks again.

**Comment by andrewconnell** [comment]
IMHO, I wouldn't invest in Angular Elements + SPFx. I've blogged why (_linked from the link in your OP_). It's just not a good dev experience as it's slow, requiring you to use two projects. This makes it a slow dev cycle.

Speaking as someone who used to be an Angular person, I've switched over to React as it's much less opinionated how things are done and is easier to use independently as well as within other solutions (such as SPFx).
